### PR TITLE
[show][config] fix for show/config muxcable hwmode model value; fix show/config muxcable return codes;

### PR DESCRIPTION
**Pull request by vdahiya12**
#### Summary

This PR fixes the 
```
show mux hwmode muxdirection
config mux hwmode state
```
cli commands to correctly use the model values for matching the vendor model value present in TRANSCEIVER_INFO table
in state DB.
                  
This PR also supports the fix for return codes for 
```
show mux
config mux 
```
commands to return 0 if the command executes correctly. 

#### What I did
fix for show/config muxcable hwmode model value; fix show/config muxcable return codes;

#### How I did it
cli commands fixed in show/muxcable.py and config/muxcable.py
fixed the unit tests as well

#### How to verify it
Ran the command on the 7050 testbed

#### New command output (if the output of a command-line utility has changed)

Signed-off-by: vaibhav-dahiya <vdahiya@microsoft.com>

### CODE REVIEW

File: config/muxcable.py [hunk 1]
```diff
@@ -13,11 +14,11 @@
 
 REDIS_TIMEOUT_MSECS = 0
 
-CONFIG_SUCCESSFUL = 100
+CONFIG_SUCCESSFUL = 0
 CONFIG_FAIL = 1
 
 VENDOR_NAME = "Credo"
-VENDOR_MODEL = "CAC125321P2PA0MS"
+REG_EXP = re.compile("CACL.+321P2P.+MS")
```
Comment (jleveque) [inline config/muxcable.py:21]: Suggest naming `VENDOR_MODEL_REGEX`

File: show/muxcable.py [hunk 2]
```diff
@@ -14,15 +15,15 @@
...
 CONFIG_FAIL = 1
 EXIT_FAIL = 1
 EXIT_SUCCESS = 0
 STATUS_FAIL = 1
-STATUS_SUCCESSFUL = 102
+STATUS_SUCCESSFUL = 0
 
 VENDOR_NAME = "Credo"
-VENDOR_MODEL = "CAC125321P2PA0MS"
+REG_EXP = re.compile("CACL.+321P2P.+MS")
```
Comment (jleveque) [inline show/muxcable.py:26]: Suggest naming `VENDOR_MODEL_REGEX`

**Review by jleveque — CHANGES_REQUESTED** [review]

File: show/muxcable.py (hunk 2 above)
Comment (vdahiya12) [reply show/muxcable.py]: fixed

File: config/muxcable.py (hunk 1 above)
Comment (vdahiya12) [reply config/muxcable.py]: fixed

File: config/muxcable.py [hunk 3]
```diff
@@ -13,11 +14,11 @@
 
 REDIS_TIMEOUT_MSECS = 0
 
-CONFIG_SUCCESSFUL = 100
+CONFIG_SUCCESSFUL = 0
 CONFIG_FAIL = 1
 
 VENDOR_NAME = "Credo"
-VENDOR_MODEL = "CAC125321P2PA0MS"
+VENDOR_MODEL_REGEX = re.compile("CAC.+321P2P.+MS")
```
Comment (jleveque) [inline config/muxcable.py:21]: Maybe we can tighten this regex up a bit, like `"CAC\w{3}321P2P\w{2}MS"`?

File: show/muxcable.py [hunk 4]
```diff
@@ -14,15 +15,15 @@
...
 CONFIG_FAIL = 1
 EXIT_FAIL = 1
 EXIT_SUCCESS = 0
 STATUS_FAIL = 1
-STATUS_SUCCESSFUL = 102
+STATUS_SUCCESSFUL = 0
 
 VENDOR_NAME = "Credo"
-VENDOR_MODEL = "CAC125321P2PA0MS"
+VENDOR_MODEL_REGEX = re.compile("CAC.+321P2P.+MS")
```
Comment (jleveque) [inline show/muxcable.py:26]: Maybe we can tighten this regex up a bit, like `"CAC\w{3}321P2P\w{2}MS"`?

**Review by jleveque — CHANGES_REQUESTED** [review]

File: show/muxcable.py (hunk 4 above)
Comment (vdahiya12) [reply show/muxcable.py]: fixed

File: config/muxcable.py (hunk 3 above)
Comment (vdahiya12) [reply config/muxcable.py]: fixed

File: config/muxcable.py [hunk 5]
```diff
@@ -13,11 +14,11 @@
 
 REDIS_TIMEOUT_MSECS = 0
 
-CONFIG_SUCCESSFUL = 100
+CONFIG_SUCCESSFUL = 0
 CONFIG_FAIL = 1
 
 VENDOR_NAME = "Credo"
-VENDOR_MODEL = "CAC125321P2PA0MS"
+VENDOR_MODEL_REGEX = re.compile("CAC\w{3}321P2P\w{2}MS")
```
Comment (jleveque) [inline config/muxcable.py:21]: Suggest making this a raw string: `re.compile(r"CAC\w{3}321P2P\w{2}MS")`

File: show/muxcable.py [hunk 6]
```diff
@@ -14,15 +15,15 @@
...
 CONFIG_FAIL = 1
 EXIT_FAIL = 1
 EXIT_SUCCESS = 0
 STATUS_FAIL = 1
-STATUS_SUCCESSFUL = 102
+STATUS_SUCCESSFUL = 0
 
 VENDOR_NAME = "Credo"
-VENDOR_MODEL = "CAC125321P2PA0MS"
+VENDOR_MODEL_REGEX = re.compile("CAC\w{3}321P2P\w{2}MS")
```
Comment (jleveque) [inline show/muxcable.py:26]: Suggest making this a raw string: `re.compile(r"CAC\w{3}321P2P\w{2}MS")`

**Review by jleveque — CHANGES_REQUESTED** [review]

File: show/muxcable.py (hunk 6 above)
Comment (vdahiya12) [reply show/muxcable.py]: fixed

File: config/muxcable.py (hunk 5 above)
Comment (vdahiya12) [reply config/muxcable.py]: fixed

File: config/muxcable.py [hunk 7]
```diff
@@ -333,7 +334,7 @@ def state(state, port):
         or not. The check gives a way to differentiate between non Y cable ports and Y cable ports.
         TODO: this should be removed once their is support for multiple vendors on Y cable"""
 
-        if vendor_value != VENDOR_NAME or model_value != VENDOR_MODEL:
+        if vendor_value != VENDOR_NAME or bool(re.match(VENDOR_MODEL_REGEX, model_value)) is False:
```
Comment (jleveque) [inline config/muxcable.py:337]: Suggest simplifying:

```python
if vendor_value != VENDOR_NAME or not re.match(VENDOR_MODEL_REGEX, model_value):
```

File: config/muxcable.py [hunk 8]
```diff
@@ -419,7 +420,7 @@ def state(state, port):
             or not. The check gives a way to differentiate between non Y cable ports and Y cable ports.
             TODO: this should be removed once their is support for multiple vendors on Y cable"""
 
-            if vendor_value != VENDOR_NAME or model_value != VENDOR_MODEL:
+            if vendor_value != VENDOR_NAME or bool(re.match(VENDOR_MODEL_REGEX, model_value)) is False:
```
Comment (jleveque) [inline config/muxcable.py:423]:
```python
if vendor_value != VENDOR_NAME or not re.match(VENDOR_MODEL_REGEX, model_value):
```

File: show/muxcable.py [hunk 9]
```diff
@@ -489,7 +490,7 @@ def muxdirection(port):
         or not. The check gives a way to differentiate between non Y cable ports and Y cable ports.
         TODO: this should be removed once their is support for multiple vendors on Y cable"""
 
-        if vendor_value != VENDOR_NAME or model_value != VENDOR_MODEL:
+        if vendor_value != VENDOR_NAME or bool(re.match(VENDOR_MODEL_REGEX, model_value)) is False:
```
Comment (jleveque) [inline show/muxcable.py:493]:
```python
if vendor_value != VENDOR_NAME or not re.match(VENDOR_MODEL_REGEX, model_value):
```

File: show/muxcable.py [hunk 10]
```diff
@@ -585,7 +586,7 @@ def muxdirection(port):
             or not. The check gives a way to differentiate between non Y cable ports and Y cable ports.
             TODO: this should be removed once their is support for multiple vendors on Y cable"""
 
-            if vendor_value != VENDOR_NAME or model_value != VENDOR_MODEL:
+            if vendor_value != VENDOR_NAME or bool(re.match(VENDOR_MODEL_REGEX, model_value)) is False:
```
Comment (jleveque) [inline show/muxcable.py:589]:
```python
if vendor_value != VENDOR_NAME or not re.match(VENDOR_MODEL_REGEX, model_value):
```

**Review by jleveque — CHANGES_REQUESTED** [review]

File: config/muxcable.py (hunk 7 above)
Comment (vdahiya12) [reply config/muxcable.py]: fixed

File: config/muxcable.py (hunk 8 above)
Comment (vdahiya12) [reply config/muxcable.py]: fixed

File: show/muxcable.py (hunk 10 above)
Comment (vdahiya12) [reply show/muxcable.py]: fixed

File: show/muxcable.py (hunk 9 above)
Comment (vdahiya12) [reply show/muxcable.py]: fixed

**Review by jleveque — APPROVED** [review]

**Comment by vdahiya12** [comment]
Retest this please

**Pull request merged**